File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums)
    {
        TreeSet<Integer>ts=new TreeSet<>();
        for(int x:nums)
            ts.add(x);
        int prev=0,count=0,mxcount=0;
        boolean first=false;
        for(Integer x:ts)
        {
            if(first==false)
            {
                prev=x;
                first=true;
                count=1;
                mxcount=1;
            }
            else
            {
                if(x==prev+1)
                {
                    count++;
                    mxcount=Math.max(mxcount,count);
                }
                else
                {
                    count=1;
                }
                prev=x;
            }
        }    
        return mxcount;
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) 
    {
        map<int,int>m;
        int c=0,s=0;
        m[0]=1;
        for(int i=0;i<nums.size();i++)
        {
            s+=nums[i];
            
            //if(s==k)
            //    c++;
            if(m.find(s-k)!=m.end())
                c+=m[s-k];
            m[s]++;
        }    
        return c;
    }
};